src/EditorFeatures/Core/CommandHandlers/QuickInfoCommandHandlerAndSourceProvider.cs

```diff
@@ -16,65 +16,31 @@
 using Microsoft.VisualStudio.Text.Editor.Commanding.Commands;
 using Microsoft.VisualStudio.Utilities;
 
-#pragma warning disable CS0618 // IQuickInfo* is obsolete, tracked by https://github.com/dotnet/roslyn/issues/24094
 namespace Microsoft.CodeAnalysis.Editor.CommandHandlers
 {
     [Export]
     [Order(After = PredefinedQuickInfoPresenterNames.RoslynQuickInfoPresenter)]
     [ContentType(ContentTypeNames.RoslynContentType)]
-    [Export(typeof(IQuickInfoSourceProvider))]
+    [Export(typeof(IAsyncQuickInfoSourceProvider))]
     [Name("RoslynQuickInfoProvider")]
     internal partial class QuickInfoCommandHandlerAndSourceProvider :
```

Comment:
This name no longer seems correct. This class is only constructing an IAsyncQuickInfoSource/Provider and no longer offers up a command handler for the InvokeQuickInfoCommand. I think it should probably be renamed to QuickInfoSourceProvider and moved to Core\Intellisense\QuickInfo.

Reply:
Ah, my bad for not explaining. I guess my statement was more directed to @gundermanc anyways.

Imagine you have quick info, completion, and signature help all up at once. You hit escape. Which one wins? We implemented this in Roslyn via having that AbstractIntellisenseCommandHandler hand the command of to each in order. (Why did we no have three separate command handlers with proper ordering between them? I don't remember. @CyrusNajmabadi might.)

Now that we're letting the editor do all of this, that code can go away, but @gundermanc has to make sure he got this right.

Reply:
We handle escape ordering for all popups and changes in context now via the IObscuringTipManager in the editor which keeps track of the dismiss order of all tooltips, intellisense features, peek, find, and other obscuring adornments. It should no longer be necessary for Roslyn to handle these behaviors.

Reply:
More specifically, we have a stack of editor features that dismisses the most recently invoked feature on esc.